NDTensors/src/imports.jl

```diff
@@ -63,6 +62,8 @@ using Base.Threads: @spawn
 using .CUDAExtensions: cu
 using .MetalExtensions: mtl
 using .GPUArraysCoreExtensions: cpu
+## Adding this here for now so that ndims in `NDTensors`` uses `TypeParameterAccessors`
+using .TypeParameterAccessors: ndims
```

Comment:
Potentially I shouldn't have this here and I should be more explicit? This is to allow `NDTensors.ndims` to still be functional.

Reply:
I would say let's remove this, if users want to use `TypeParameterAccessors.ndims` they should be explicit about that. `NDTensors.cu` is a different situation because we have recommended that to users in discourse posts.

Reply:
Perhaps we should establish some rules about using `ndims`, since it seems like it could get quite complicated otherwise. One proposal would be the following:
1. Bias towards using `Base.ndims` throughout the code.
2. Only use `TypeParameterAccessors.ndims` if needed to circumvent an issue with `Base.ndims`.
3. When using `Base.ndims`, write it in the code as `ndims`.
4. When using `TypeParameterAccessors.ndims`, type it explicitly as `TypeParameterAccessors.ndims`, not `ndims`.

So we should think about `TypeParameterAccessors.ndims` as a workaround for issues with `Base.ndims` that we use sparingly. Otherwise, I think it will get chaotic trying to decide when to use `TypeParameterAccessors.ndims` or `Base.ndims`, what `ndims` is supposed to mean, etc.

Reply:
An exception to the rules above may be that within the `TypeParameterAccessors` module itself, we may need to use `Base.ndims` explicitly to avoid having Julia interpret `ndims` as `TypeParameterAccessors.ndims`.

Reply:
Also note that we don't have to update the entire codebase with these new rules/conventions in this PR, maybe just make sure that the code that you are changing otherwise in this PR follows those rules. It's not really worth updating all of `NDTensors`, and instead focus on using those rules consistently in the new library submodules.

Reply:
It looks like we actually don't even need our own version of `ndims`. I removed it from the codebase and all tests pass in ITensors and NDTensors. I guess the question is, do we still want a version in `TypeParameterAccessors` even if its not used?

Reply:
If I recall correctly it was defined to circumvent this issue in older versions of Julia:
```julia
julia> ndims(Array{<:Any,3})
ERROR: MethodError: no method matching ndims(::Type{Array{var"#s63", 3} where var"#s63"})
Closest candidates are:
  ndims(::CartesianIndices) at multidimensional.jl:362
  ndims(::Base.Iterators.ProductIterator) at iterators.jl:967
  ndims(::AbstractArray{T, N}) where {T, N} at abstractarray.jl:212
  ...
Stacktrace:
 [1] top-level scope
   @ REPL[3]:1

julia> versioninfo()
Julia Version 1.6.7
Commit 3b76b25b64 (2022-07-19 15:11 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin21.4.0)
  CPU: Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, westmere)
```
while in more recent versions:
```julia
julia> ndims(Array{<:Any,3})
3

julia> versioninfo()
Julia Version 1.10.2
Commit bd47eca2c8a (2024-03-01 10:14 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 10 × Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
Threads: 1 default, 0 interactive, 1 GC (on 8 virtual cores)
```

Reply:
I completely agree that is why we had implemented it in the past, but I think the new design we circumvent times when we would have previously called ndims using the new `set_ndims` function and using the function like `type_parameters(type, (eltype, ndims))`. So now all unittests pass using the current Julia version and V1.6. I double checked on my machine and found all 1.6 test pass

Reply:
I see, thanks for the explanation. So if I understand correctly, effectively instead of having `TypeParameterAccessors.ndims`, if `Base.ndims(type)` has an issue in an older version of Julia, we can use `type_parameter(type, ndims)`. That sounds reasonable, let's get rid of `NDTensors.ndims`/`TypeParameterAccessors.ndims` and go with that design.

Reply:
Yes that is correct and seems safer/more explicit.  The `NDTensors` and `TypeParameterAccessors` `ndims` function is already removed and this PR should be ready for review.